tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs

```diff
@@ -6,158 +6,155 @@
                 }
             }
-
-            return new CallTargetState(null, new object[] { testSuite, message });
         }
 
-        /// <summary>
-        /// OnMethodEnd callback
-        /// </summary>
-        /// <typeparam name="TTarget">Type of the target</typeparam>
-        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
-        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
-        /// <param name="state">Calltarget state value</param>
-        /// <returns>Return value of the method</returns>
-        internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exception exception, in CallTargetState state)
+        return new CallTargetState(null, new object[] { testSuite, message });
+    }
+
+    /// <summary>
+    /// OnMethodEnd callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+    /// <param name="state">Calltarget state value</param>
+    /// <returns>Return value of the method</returns>
+    internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exception exception, in CallTargetState state)
+    {
+        if (state.State != null)
         {
-            if (state.State != null)
+            var stateArray = (object[])state.State;
+            var testSuiteOrWorkItem = stateArray[0];
+            var skipMessage = (string)stateArray[1];
+
+            const string startString = "OneTimeSetUp:";
+            if (skipMessage?.StartsWith(startString, StringComparison.OrdinalIgnoreCase) == true)
             {
-                object[] stateArray = (object[])state.State;
-                object testSuiteOrWorkItem = stateArray[0];
-                string skipMessage = (string)stateArray[1];
+                skipMessage = skipMessage.Substring(startString.Length).Trim();
+            }
 
-                const string startString = "OneTimeSetUp:";
-                if (skipMessage?.StartsWith(startString, StringComparison.OrdinalIgnoreCase) == true)
-                {
-                    skipMessage = skipMessage.Substring(startString.Length).Trim();
-                }
+            if (testSuiteOrWorkItem is not null)
+            {
+                var typeName = testSuiteOrWorkItem.GetType().Name;
 
-                if (testSuiteOrWorkItem is not null)
+                if (typeName == "ParameterizedMethodSuite")
                 {
-                    string typeName = testSuiteOrWorkItem.GetType().Name;
-
-                    if (typeName == "ParameterizedMethodSuite")
+                    // In case the TestSuite is a ParameterizedMethodSuite instance
+                    var testSuite = testSuiteOrWorkItem.DuckCast<ITestSuite>();
```

Comment:
Should we replace `ITestSuite` with `ITest` now that the `ITestSuite` duck type doesn't have any additional proxy members?

```suggestion
                    var testSuite = testSuiteOrWorkItem.DuckCast<ITest>();
```

Reply:
I prefer to keep it separated, conceptually improves readibility and also it have a performance gain due the way DuckType works (https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/DuckTyping/DuckType.cs#L1166)